addons/sale/models/sale.py

```diff
@@ -158,6 +158,13 @@ def _inverse_project_id(self):
 
     product_id = fields.Many2one('product.product', related='order_line.product_id', string='Product')
 
+    def _search_invoice_ids(self, operator, value):
+        invoice_ids = []
+        for order in self:
+            invoice_ids += order.order_line.mapped('invoice_lines').mapped('invoice_id').filtered(lambda r: r.type in ['out_invoice', 'out_refund']).mapped('id')
+        return [('id', 'in', invoice_ids)]
+
+
```

Comment:
@robyf70 you can use dote in mapped fonction. 
You can write something more compact:
```
def _search_invoice_ids(self, operator, value):
     invoices = self.mapped('order_line.invoice_lines.invoice_id').filtered(lambda r: r.type in ['out_invoice', 'out_refund'])
     return [('id', 'in', invoices.ids)]
```

regards

Reply:
Right! Will do that!